### ISSUE TITLE
Use Option<PFN>

### ISSUE THREAD

**Issue by TimDiekmann** [Osspial/vk-rs#15]
Is it possible to use `Option<PFN>` where the `vk.xml` states function pointers as optional? An example would be `VkAllocationCallbacks`:

```rust
#[repr(C)]
pub struct VkAllocationCallbacks {
	pub p_user_data:             *mut c_void,
	pub pfn_allocation:          PFN_vkAllocationFunction,
	pub pfn_reallocation:        PFN_vkReallocationFunction,
	pub pfn_free:                PFN_vkFreeFunction,
	pub pfn_internal_allocation: Option<PFN_vkInternalAllocationNotification>,
	pub pfn_internal_free:       Option<PFN_vkInternalFreeNotification>,
}
```

[This](https://play.rust-lang.org/?gist=6d5638b8bc14b0db9f0df8e7e2d07cee&version=stable) will never fail due to [Null Pointer Optimization](https://doc.rust-lang.org/book/first-edition/ffi.html#the-nullable-pointer-optimization):
```rust
assert_eq!(size_of::<PFN>(), size_of::<Option<PFN>>());
```

**Comment by Osspial** [comment]
The parser already keeps track of which members of structs are optional, but the generator currently does nothing with that information. However, it should be pretty easy to take account of, and I'll look into adding this in.

**Comment by Osspial** [comment]
Fixed in 2845ba369aa40346e068bb192a342ba79c24c378 and 3a4c2268485a5196f41c4b8214fa67cb70b1d294
